pkg/client/auditor/auditor.go

```diff
@@ -299,6 +296,12 @@ func (a *defaultAuditor) audit() error {
 			return noErr
 		}
 
+		if err := a.verifyStateSignature(serverID, prevState); err != nil {
```

Comment:
I think this additional signature verification is not needed. Verifying signature before accepting state evolution should be enough.
Verification of local state will double the amount of signature verifications and will make it harder to deal with changing server signing keys. If verification only happens when a new state is validated, it's easier to deal with a new key by just changing the public key at the auditor.

Reply:
Fixed: i removed the signature verification for the previous state.